src/web/src/pages/Main.vue

```diff
@@ -118,20 +118,11 @@ export default {
       ICS_DAY_SHORTNAMES: ['MO', 'TU', 'WE', 'TH', 'FR', 'SA', 'SU']
     };
   },
-  created() {
-    if (this.$route.query.semester) {
-      this.currentSemester = this.$route.query.semester;
-    } else {
-      getDefaultSemester().then(semester => {
-        this.currentSemester = semester;
-      });
-    }
+  async created() {
+    this.currentSemester = this.$route.query.semester ? this.$route.query.semester : await getDefaultSemester();
 
-    getCourses().then(courses => {
+    Promise.all([getCourses(this.currentSemester), getSubSemesters(this.currentSemester)]).then(([courses, subsemesters]) => {
```

Comment:
If the function is already `async` why opt for `Promise.all` over separate `await`s?

Reply:
await will halt the execution of the main thread until that function has finished, no? Promise.all allows these actions to occur async at the same time

Reply:
True, I guess its a readability vs optimization thing